Assets/Assets/Scripts/MoveController.cs

```diff
@@ -0,0 +1,260 @@
+    }
+
+    /// <summary>
+    /// Method
+    /// Selects a piece and returns its legal moves.
+    /// If a multicapture is in progress, this method makes no updates,
+    /// and continues selecting the multicapturing piece.
+    /// </summary>
+    public List<CheckersMove.Move> SelectPiece(CheckersMove.Square square)
+    {
+        if (isMulticaptureInProgress) return legalMoves;
+
+        selectedSquare = square;
+
+        if(selectedSquare is CheckersMove.Square validSquare)
+        {
+            legalMoves = LegalMoveGenerator.GetLegalMoves(validSquare, boardState, currentTurn, forceCaptures);
+        }
+        else
+        {
+            selectedSquare = null;
+            legalMoves.Clear();
+        }
+
+        return legalMoves;
+    }
+
+    /// <summary>
+    /// Method
+    /// Deselects a piece and resets the legal moves list.
+    /// If a multicapture is in progress, this method makes no updates,
+    /// and continues selecting the multicapturing piece.
+    /// </summary>
+    public void DeselectPiece()
+    {
+        if(isMulticaptureInProgress) return;
+        selectedSquare = null;
+    }
+
+    /// <summary>
+    /// Method
+    /// Attempts to move the currently selected piece to the square passed in.
+    /// Will return true and perform the move if and only if the move is legal.
+    /// Alsp detects if a multicapture is available.
+    /// </summary>
+    public bool MakeMove(CheckersMove.Square destSquare)
+    {
+        if(selectedSquare is CheckersMove.Square srcSquare)
+        {
+            CheckersMove.Move move = new CheckersMove.Move(srcSquare, destSquare);
+            if(legalMoves.Contains(move))
+            {
+                List<CheckersMove.Move> multicaptures = LegalMoveGenerator.MakeLegalMove(move, ref boardState, currentTurn);
+                isMulticaptureInProgress = false;
+
+                if(multicaptures.Count > 0)
+                {
+                    SelectPiece(move.dest);
+                    isMulticaptureInProgress = true;
+                    legalMoves = multicaptures;
+                }
+                else
+                {
+                    isMulticaptureInProgress = false;
+                    SwapTurns();
+                }
+
+                return true;
+            }
+        }
+        
+        return false;
+    }
+
+    /// <summary>
+    /// Method
+    /// Declines the option to multicapture and updates state accordingly.
+    /// Multicaptures cannot be delclined if capturing is forced; in that case,
+    /// the method does nothing.
+    /// </summary>
+    public void DeclineMulticapture()
+    {
+        if(!forceCaptures && isMulticaptureInProgress)
+        {
+            isMulticaptureInProgress = false;
+            SwapTurns();
+        }
+    }
+
+    /// <summary>
+    /// Method
+    /// Helper method for handling state change upon switching turns.
+    /// </summary>
+    private void SwapTurns()
```

Comment:
Something regarding changing turns isn't allowing for multi-capture to occur

Reply:
Could you give steps to reproduce the error?

Reply:
Black d6 -> e5
White g3 -> f4
Black b6 -> c5
White a3 -> b4
Black a7 -> b6
White b2 -> a3
Black b8 -> a7
White f4-> d6
Cannot White -> b8 because its Black's turn, therefore Black c7 -> e5